src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -448,8 +458,8 @@ public HttpClientSettingsKey getHttpClientKey() throws SnowflakeSQLException {
                   httpProxyHost,
                   proxyPort,
                   combinedNonProxyHosts,
-                  "", /* user = empty */
-                  "", /* password = empty */
+                  httpProxyUser,
+                  httpProxyPassword,
```

Comment:
General question - this covers only giving proxy settings from JVM parameters, right? How about passing such parameters in connection string?